DashWallet/DWTxHistoryViewController.m

```diff
@@ -593,6 +526,208 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
+        unconfirmedLabel.text = NSLocalizedString(@"locked", nil);
+        unconfirmedLabel.backgroundColor = [UIColor colorWithWhite:0.0 alpha:0.2];
+        textLabel.textColor = [UIColor yellowColor];
+        balanceLabel.text = localBalanceLabel.text = nil;
+    }
+    else if (!instantSendReceived && confirms < 6) {
+        if (confirms == 0) unconfirmedLabel.text = NSLocalizedString(@"0 confirmations", nil);
+        else if (confirms == 1) unconfirmedLabel.text = NSLocalizedString(@"1 confirmation", nil);
+        else unconfirmedLabel.text = [NSString stringWithFormat:NSLocalizedString(@"%d confirmations", nil),
+                                      (int)confirms];
+    }
+    else {
+        unconfirmedLabel.text = nil;
+        unconfirmedLabel.hidden = YES;
+        sentLabel.hidden = NO;
+    }
+    sentLabel.textColor = [UIColor whiteColor];
+    if (sent > 0 && received == sent) {
+        textLabel.attributedText = [priceManager attributedStringForDashAmount:sent withTintColor:textLabel.textColor];
+        localCurrencyLabel.text = [NSString stringWithFormat:@"(%@)",
+                                   [priceManager localCurrencyStringForDashAmount:sent]];
+        sentLabel.text = NSLocalizedString(@"Moved", nil);
+        sentLabel.backgroundColor = UIColorFromRGB(0x008DE4);
+    }
+    else if (sent > 0) {
+        textLabel.attributedText = [priceManager attributedStringForDashAmount:received - sent withTintColor:textLabel.textColor];
+        localCurrencyLabel.text = [NSString stringWithFormat:@"(%@)",
+                                   [priceManager localCurrencyStringForDashAmount:received - sent]];
+        sentLabel.text = NSLocalizedString(@"Sent", nil);
+        sentLabel.backgroundColor = UIColorFromRGB(0xD0021B);
+    }
+    else {
+        textLabel.attributedText = [priceManager attributedStringForDashAmount:received withTintColor:textLabel.textColor];
+        localCurrencyLabel.text = [NSString stringWithFormat:@"(%@)",
+                                   [priceManager localCurrencyStringForDashAmount:received]];
+        if (instantSendReceived) {
+            sentLabel.text = NSLocalizedString(@"Received", nil);
+        } else {
+            sentLabel.text = NSLocalizedString(@"Received", nil);
+        }
+        sentLabel.backgroundColor = UIColorFromRGB(0x7ED321);
+    }
+    
+    if (! unconfirmedLabel.hidden) {
+        unconfirmedLabel.layer.cornerRadius = 9.0;
+        unconfirmedLabel.text = [unconfirmedLabel.text stringByAppendingString:@"   "];
+        unconfirmedLabel.backgroundColor = [UIColor colorWithWhite:0.85 alpha:1.0];
+    }
+    else {
+        sentLabel.layer.cornerRadius = 9.0;
+        sentLabel.text = [sentLabel.text stringByAppendingString:@"   "];
+        sentLabel.highlightedTextColor = sentLabel.textColor;
+    }
+}
+
+#if SNAPSHOT
+- (void)mock_configureTxCell:(DWTransactionTableViewCell *)transactionCell indexPath:(NSIndexPath *)indexPath {
```

Comment:
I would suggest we use configureTxCell as well for the mock transactions and have transactions and mock transactions use the same protocol.